### PR TITLE
Provisioning scripts

### PR DESCRIPTION
**Pull request by mosabua**
Issue https://issues.sonatype.org/browse/NEXUS-9758

CI build at http://bamboo.s/browse/BOOK-NEXUS95

References example scripts are in https://github.com/sonatype/nexus-book-examples/tree/nexus-3.0.x/scripting - please also review those. Feel free to commit improvements there straight to the branch.

### CODE REVIEW

File: chapter-scripting.asciidoc [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+a Maven hosted repository in the simplest usage simply requires a name.
+
+----
+repository.createMavenHosted("private")
+----
+
+This method simply uses the default values for the rest of the parameters and is therefore equivalent to
+
+----
+repository.createMavenHosted("private",
```
Comment (kellyrob99) [inline chapter-scripting.asciidoc]: incomplete by design still?

File: chapter-scripting.asciidoc [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+A list of scripts stored on the repository manager can be accessed with
+
+----
+curl -v -X GET -u admin:admin123 'http://localhost:8081/service/siesta/rest/v1/script'
+----
+
+A script can subsequently be executed by sending a POST to the `run` method of the specific script
+
+----
+curl -v -X POST -u admin:admin123 --header "Content-Type: text/plain" 'http://localhost:8081/service/siesta/rest/v1/script/maven/run'
```
Comment (kellyrob99) [inline chapter-scripting.asciidoc]: this also accepts a String input so you can pass arguments to the script

**Comment by mosabua** [comment]
Should we reference the scripting chapter from the installation and other sections yet @kellyrob99 ?

File: chapter-scripting.asciidoc [hunk 3]
```diff
@@ -0,0 +1,210 @@
...
+.Example JSON formatted file `maven.json` with a simple repository creation script 
+----
+{
+  "name": "maven",
+  "type": "groovy",
+  "content": "repository.createMavenHosted('private')"
+}
+----
+
+The JSON file can be published to the repository manager with a HTTP POST operation like
```
Comment (kellyrob99) [inline chapter-scripting.asciidoc]: "an" not "a" HTTP POST

Comment (kellyrob99) [inline chapter-scripting.asciidoc]: Should explicitly mention here that maven.json is a file in the same directory?

File: chapter-scripting.asciidoc [hunk 4]
```diff
@@ -0,0 +1,210 @@
...
+
+////
+Note: the nexus-testsuite includes a number of example scripts,
+ unfortunately they are not available to the public so we need to duplicate them here for now
+
+https://github.com/sonatype/nexus-internal/tree/master/private/testsuite/nexus-testsuite-data/src/main/resources/provisioningScripts
+
+////
+
+The API for scripts is capable of a number of different task. This section provides examples for script writing, publishing
```
Comment (kellyrob99) [inline chapter-scripting.asciidoc]: should be "tasks" plural here

File: chapter-scripting.asciidoc (hunk 3 above)
Comment (mosabua) [inline chapter-scripting.asciidoc]: Both fixed

File: chapter-scripting.asciidoc (hunk 4 above)
Comment (mosabua) [inline chapter-scripting.asciidoc]: Fixed.

**Comment by mosabua** [comment]
Imho this is now ready to go..

**Comment by mosabua** [comment]
@kellyrob99 agreed this is ready for merge. Any further changes can be done separately where necessary.

**Pull request merged**